Framework/NexusCpp/inc/MantidNexusCpp/NeXusFile_fwd.h

```diff
@@ -0,0 +1,137 @@
+/** This class defines data types which are used as part of the NeXus API.
+ * They should more properly be moved into NeXusFile, when the nexus layer has been cleaned up.
+ * OR all type and enum definitions in NeXusFile all moved here.
+ */
+
+#pragma once
+
+#include <iostream>
+
+typedef const char CONSTCHAR;
+
+typedef void *NXhandle; /* really a pointer to a NexusFile structure */
+typedef char NXname[128];
+
+/** \enum NXaccess_mode
+ * NeXus file access codes.
+ * \li NXACC_READ read-only
+ * \li NXACC_RDWR open an existing file for reading and writing.
+ * \li NXACC_CREATE create a NeXus HDF-4 file
+ * \li NXACC_CREATE4 create a NeXus HDF-4 file
```

Comment:
Add a comment that this is discouraged because it will soon no longer be supported.